assets/blocks/button/color-hooks.js

```diff
@@ -0,0 +1,124 @@
+import { addFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+import { omitBy } from 'lodash';
+import { withColorSettings } from '../../shared/blocks/settings';
+
+/**
+ * Check if block is a Sensei button.
+ *
+ * @param {Object|string} blockType Block settings or name.
+ * @return {boolean} Is Sensei button.
+ */
+const isSenseiButton = ( blockType ) => {
+	blockType = 'string' === typeof blockType ? blockType : blockType.name;
+	return blockType && blockType.startsWith( 'sensei-lms/button' );
+};
+
+/**
+ * Add fallback Color settings and attributes if the color support key is not available.
+ *
+ * @param {Object} settings Block settings.
+ * @return {Object} Block settings.
+ */
+export const addColorSettings = ( settings ) => {
+	if ( ! isSenseiButton( settings ) ) return settings;
+
+	if ( settings.supports?.color && ! settings.attributes?.backgroundColor ) {
+		settings.edit = withColorSettings( {
```

Comment:
If the `supports: {color: true}` option is supported by current Gutenberg, it will have already added `backgroundColor` (and a few others here). If it's not there, wrap the edit component it a HOC that hooks up the color settings panel and attributes, and add the attributes needed. (Which are a bit different, having an extra `custom*` one for each color)